iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -0,0 +1,227 @@
+
+                try
+                {
+                    await SendMessageAsync(count);
+                }
+                catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
+                {
+                    // Inspect the exception to figure out if operation should be retried, or if user-input is required.
+                    _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
+                }
+                catch (Exception ex)
+                {
+                    if (ExceptionHelper.IsNetworkExceptionChain(ex))
+                    {
+                        _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+
+                await Task.Delay((int)s_sleepDuration.TotalMilliseconds);
+            }
+        }
+
+        private async Task ReceiveMessagesAsync()
+        {
+            while (true)
+            {
+                _logger.LogInformation($"Device waiting for C2D messages from the hub - for {s_sleepDuration}...");
+                _logger.LogInformation("Use the IoT Hub Azure Portal or Azure IoT Explorer to send a message to this device.");
+
+                try
+                {
+                    await ReceiveMessageAndCompleteAsync();
+                }
+                catch (Exception ex) when (ex is DeviceMessageLockLostException)
+                {
+                    _logger.LogWarning($"Attempted to complete a received message whose lock token has expired; ignoring: {ex}");
+                }
+                catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
+                {
+                    // Inspect the exception to figure out if operation should be retried, or if user-input is required.
+                    _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
+                }
+                catch (Exception ex)
+                {
+                    if (ExceptionHelper.IsNetworkExceptionChain(ex))
+                    {
+                        _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+            }
+        }
+
+        private async Task SendMessageAsync(int count)
+        {
+            var _temperature = s_randomGenerator.Next(20, 35);
+            var _humidity = s_randomGenerator.Next(60, 80);
+            string dataBuffer = $"{{\"temperature\":{_temperature},\"humidity\":{_humidity}}}";
+
+            using var eventMessage = new Message(Encoding.UTF8.GetBytes(dataBuffer))
```

Comment:
Add content encoding and content type properties?

Reply:
sure, can do